### PR TITLE
Allow me to create a `Custom Constraint Class` and use it in the same file (`filepath=None`)

### LINKED ISSUE #1205: Allow me to create a custom constraint class and use it in the same file
### Problem Description
The current API assumes that the custom constraint class is defined a Python file, and then imported/created in a different Python file. I would like the option to create a class and use it in the same file

_This also makes demoing & internally testing a custom constraint a lot easier._

### Expected behavior
In the new SDV 1.0 API, the workflow is changing a bit.  When loading a constraint class from a different file, we will offer this function:

```python
synthesizer.load_custom_constraint_classes(
    filepath='my_custom_constraint.py',
    class_names=['MyCustomConstraint'])
```

When you want to do it from the same file, you will have the object. We should add a new function to add it.

```python
synthesizer.add_custom_constraint_class(
    class_object=MyCustomConstraint,
    class_name='MyCustomConstraint' # the name to use when referring to the constraint
)
```

Then you can create constraints using the defined `class_name`:

```python
constraint = {
  'constraint_class': 'class_name',
  'constraint_parameters': {
    'column_names': ['A', 'B'],
    'my_custom_parameter': 10
  }
}
```

### PR DESCRIPTION
**Pull request by amontanez24**
resolves #1205

### CODE REVIEW

**Review by pvk-developer — CHANGES_REQUESTED** [review]
Just a few comments, but then it should be good!
Also I realized there are no integration tests for `multi_table` for both `load` and `add` do you mind adding those ? Or we can address it later.

File: sdv/data_processing/data_processor.py [hunk 1]
```diff
@@ -180,9 +179,24 @@ def load_custom_constraint_classes(self, filepath, class_names):
...
+        """Add a custom constraint class for the synthesizer to use.
+
+        Args:
+            class_object (sdv.constraints.Constraint):
+                A custom constraint class object.
+            class_name (str):
+                The name to assign this custom constraint class. This will be the name to use
+                when writing a constraint dictionary for ``add_constraints``.
+        """
+        self._custom_constraint_classes[class_name] = class_object
```
[the remark this answers is not in the document]
Comment (amontanez24) [reply sdv/data_processing/data_processor.py:199]: good catch

**Review by pvk-developer — APPROVED** [review]
🚢 LG to me.

File: tests/integration/single_table/custom_constraints.py [hunk 2]
```diff
@@ -10,12 +10,14 @@ def is_valid(column_names, data):
 
 def transform(column_names, data):
     """Transform the constraint."""
-    return data[column_names] ** 2
+    data[column_names] = data[column_names] ** 2
```
Comment (fealho) [inline tests/integration/single_table/custom_constraints.py:13]: Good catch!

**Review by fealho — APPROVED** [review]
Pretty good 👍

**Pull request merged**